lms/djangoapps/course_home_api/progress/api.py

```diff
@@ -4,8 +4,200 @@
 
 from django.contrib.auth import get_user_model
 from opaque_keys.edx.keys import CourseKey
+from xmodule.graders import ShowCorrectness
+from datetime import datetime, timezone
 
 from lms.djangoapps.courseware.courses import get_course_blocks_completion_summary
+from dataclasses import dataclass, field
+from typing import Optional, List, Dict, Tuple
+
+
+@dataclass
+class _AssignmentBucket:
```

Comment:
Question: Do we use the terminology "bucket" elsewhere in the grading code? I don't hate the name, but I don't want to add a convention if we already have something else we call this, say on the frontend.